### PR TITLE
Garcia-Changes

### PR DESCRIPTION
**Pull request by fgarcia0x0**
- add big improvement's in libpe.
- add error checking for some mechanism.
- refactored code in some parts.

### CODE REVIEW

File: hashes.c [hunk 1]
```diff
@@ -32,6 +32,10 @@
 #include <math.h>
 #include <string.h>
 
+// add utility
+#define PEV_ABORT_IF(cond) \
+	do { cond ? abort() : (void)0; } while (0)
```
Comment (merces) [inline hashes.c:38]: Improvement: use `(cond)`

File: hashes.c [hunk 2]
```diff
@@ -358,18 +362,38 @@ typedef struct element {
 	struct element *next; // needed for singly- or doubly-linked lists
 } element_t;
 
+static void pev_tolower_str(char* str)
+{
+	for (char* p = str; *p; ++p)
+		*p = tolower((unsigned char)*p);
+}
+
+static void pev_get_elem_func_name(element_t* elem_ptr, ord_t* ord_ptr, int hint)
```
Comment (merces) [inline hashes.c]: We should pick up a better name.

**Review by merces — CHANGES_REQUESTED** [review]
Please review. Have a look at [these](https://www.onlinegdb.com/fork/Q1CPEpu6t) test cases.

**Comment by fgarcia0x0** [comment]
all warning and pendencies has fixed

**Comment by fgarcia0x0** [comment]
all warning and pendencies has fixed

**Review by merces — APPROVED** [review]
These are great! Thanks!

**Pull request merged**